src/Analyzers/CSharp/Analysis/RemoveRedundantToStringCallAnalysis.cs

```diff
@@ -64,10 +64,23 @@ public static void Analyze(SyntaxNodeAnalysisContext context, in SimpleMemberInv
                 if (containingType.SpecialType == SpecialType.System_String)
                     return true;
 
-                if (invocationExpression.WalkUpParentheses().IsParentKind(SyntaxKind.Interpolation)
-                    && IsNotHidden(methodSymbol, containingType))
+                ExpressionSyntax expression = invocationExpression.WalkUpParentheses();
+                switch (expression.Parent.Kind())
                 {
-                    return true;
+                    case SyntaxKind.Interpolation:
+                        {
+                            return IsNotHidden(methodSymbol, containingType);
+                        }
+                    case SyntaxKind.AddExpression:
+                        {
+                            var addExpression = (BinaryExpressionSyntax)expression.Parent;
+                            if (addExpression.Right == expression)
+                                return semanticModel.GetTypeInfo(addExpression.Left, cancellationToken).Type?.SpecialType == SpecialType.System_String;
+
+                            return semanticModel.GetTypeInfo(addExpression.Right, cancellationToken).Type?.SpecialType == SpecialType.System_String
+                                && (addExpression.Right.WalkDownParentheses() is not InvocationExpressionSyntax invocationExpression2
+                                    || semanticModel.GetMethodSymbol(invocationExpression2, cancellationToken).Name != "ToString");
```

Comment:
One more thing: It would be more precise to check if the method symbol is exactly the `ToString` we are looking for, i.e. use the same check as in the `if` condition several lines above:

```
methodSymbol?.DeclaredAccessibility == Accessibility.Public
    && !methodSymbol.IsStatic
    && !methodSymbol.IsGenericMethod
    && string.Equals(methodSymbol.Name, "ToString", StringComparison.Ordinal)
    && methodSymbol.ReturnType.SpecialType == SpecialType.System_String
    && !methodSymbol.Parameters.Any()
```